internal/rest/avi_obj_vsvip.go

```diff
@@ -119,7 +119,12 @@ func (rest *RestOperations) AviVsVipBuild(vsvip_meta *nodes.AviVSVIPNode, vsCach
 						if vsvip.Vip[0].IPAMNetworkSubnet == nil {
 							vip.IPAMNetworkSubnet = &avimodels.IPNetworkSubnet{}
 						} else {
-							vip.IPAMNetworkSubnet = vsvip.Vip[0].IPAMNetworkSubnet
+							ipamNetworkRefSplit := strings.Split(*vsvip.Vip[0].IPAMNetworkSubnet.NetworkRef, "#")
+							if len(ipamNetworkRefSplit) > 1 && ipamNetworkRefSplit[1] != vsvip_meta.VipNetworks[0].NetworkName {
```

Comment:
do we know when and why is this condition true
`ipamNetworkRefSplit[1] != vsvip_meta.VipNetworks[0].NetworkName`?

Reply:
@aaha97 It is true when we try to update the vip network with AviInfraSetting CRD object.